packages/uploadthing/test/server.test.ts

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { beforeEach, expect, expectTypeOf, it, vi } from "vitest";
+
+import { UPLOADTHING_VERSION } from "uploadthing/constants";
+
+import { createAPIRequestUrl } from "../src/internal/ut-reporter";
+import type { ActionType } from "../src/server";
+import { createRouteHandler, createUploadthing } from "../src/server";
+
+const f = createUploadthing({
+  errorFormatter: (e) => ({
+    message: e.message,
+    cause: (e.cause as Error)?.toString(),
+  }),
+});
+const router = {
+  middlewareThrows: f({ blob: {} })
+    .middleware((opts) => {
+      expect(opts.req).toBeInstanceOf(Request);
+      expectTypeOf<Request>(opts.req);
+
+      expect(opts.event).toBeUndefined();
+      expectTypeOf<undefined>(opts.event);
+
+      expect(opts.res).toBeUndefined();
+      expectTypeOf<undefined>(opts.res);
+
+      //   expect(opts.input).toBeUndefined();
```

Comment:
here we actually have a type-mismatch. we get `{}` if no input handler is set, but type says `udnefined`